### PR TITLE
Restrict extensibility of ServerResourceOwner

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Following https://github.com/digital-asset/daml/pull/8649 this commit
fixes the last open comment from https://github.com/digital-asset/daml/pull/8604
as explained in the comment https://github.com/digital-asset/daml/pull/8604#issuecomment-766620042

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/GrpcServer.scala [hunk 1]
```diff
@@ -30,16 +30,16 @@ private[apiserver] object GrpcServer {
...
       maxInboundMessageSize: Int,
-      sslContext: Option[SslContext],
-      interceptors: List[ServerInterceptor],
+      sslContext: Option[SslContext] = None,
+      interceptors: List[ServerInterceptor] = List.empty,
       metrics: Metrics,
       servicesExecutor: Executor,
       services: Iterable[BindableService],
-  ): NettyServerBuilder = {
+  ): AbstractResourceOwner[ResourceContext, Server] = {
```
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/GrpcServer.scala:42]: Isn't this the same as `ResourceOwner[Server]`?

**Review by SamirTalwar-DA — APPROVED** [review]
This is great.

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/GrpcServer.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/GrpcServer.scala:42]: Indeed, initially I had some type error there and went for the lazy route of putting there whatever the compiler told me to. 😄 Thanks for spotting this. 🙇🏻

Comment (stefanobaghino-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/GrpcServer.scala]: Addressed by https://github.com/digital-asset/daml/pull/8660/commits/b466c8ee8a7fca1b80f7ef217f14d8c2dd3e2c80

**Pull request merged**